lib/postcode_validation/use_case/validate_address.rb

```diff
@@ -35,6 +36,28 @@ def potential_address_matches
         @address_match_gateway.query(search_term: postcode,
                                      country: country)
       end
+
+      def invalid_postcode_format?
+        downcase_country = country.downcase
+        regex = send("#{downcase_country}_postcode_regex")
```

Comment:
Adding a new country, or modifying a particular country's validation won't require touching the use-case's code (which is good - less code churn).

Also allows country validation code to be long, if necessary.